### PR TITLE
feat: add Solis inverter

### PR DESCRIPTION
**Pull request by Pho3niX90**
Added support for Solis

## Description
This should not be a breaking change.
Some code were refactored, specifically regarding the hardcoded inverter brand strings. 

## Motivation and Context
Adds support for the solis, it also introduces a new `grid_voltage` to automatically mark grid offline when the voltage is 0, since solis does not provide the status. There is also a new `battery_current_direction`, which seems to be unique to solis.

## How has this been tested
On my own system

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have updated the version in `package.json` following [semver](https://semver.org/)

### CODE REVIEW

**Comment by slipx06** [comment]
Thanks for this. What did you use to reformat all the code?

File: src/index.ts [hunk 1]
```diff
@@ -1,1692 +1,3213 @@
...
-        if (config.entities?.battery_status === 'none' || !config.entities?.battery_status) {
-          batteryStateColour = 'transparent';
-          batteryStateMsg = '';
-        } else {
-          batteryStateColour = 'transparent';
-          batteryStateMsg = 'Status';
+        if (battery_current_direction != null && !isNaN(battery_current_direction)) {
+            if (inverterModel == InverterModel.Solis && battery_current_direction === 0) {
+                battery_power = -battery_power;
+            }
```
Comment (slipx06) [inline src/index.ts:479]: Is battery power always reported as a positive number for the Solis Inverter regardless of whether you are charging or discharging ?  Trying to understand why you need this?

Comment (slipx06) [reply src/index.ts:479]: Should the logic not be:

if (inverterModel == InverterModel.Solis && battery_current_direction >= 0)

Comment (Pho3niX90) [reply src/index.ts:479]: Hi, yes always a positive value, the direction has 2 states, 0 and 1. Where 0 is for discharging

**Comment by Pho3niX90** [comment]
> Thanks for this. What did you use to reformat all the code?

It might have been the combination of eslint and intellij

**Pull request merged**